cpp/include/raft/distance/detail/fused_l2_nn.cuh

```diff
@@ -380,6 +382,91 @@ void fusedL2NNImpl(OutT* min,
   }
 }
 
+template <bool sqrt, typename DataT, typename IdxT, typename LabelT>
+RAFT_KERNEL fusedL2NNKernelSmallInput(const DataT* dataset,
+                                      const DataT* centers,
+                                      const DataT* dataset_norm,
+                                      const DataT* centers_norm, 
+                                      IdxT n_rows,
+                                      IdxT n_clusters,
+                                      IdxT dim,
+                                      LabelT* labels)
+{
+  extern __shared__ char smem[];
+  DataT *centers_shared = reinterpret_cast<DataT*>(smem);
+  DataT *dataset_shared = centers_shared + n_clusters * dim; 
+  int starting_row = blockDim.x * blockIdx.x; 
+  int curr_row = blockDim.x * blockIdx.x + threadIdx.x; 
+
+  int shmem_loading_idx = threadIdx.x; 
+  while (shmem_loading_idx < n_clusters * dim) {
+    centers_shared[shmem_loading_idx] = centers[shmem_loading_idx];
+    shmem_loading_idx += blockDim.x; 
+  }
+
+  shmem_loading_idx = threadIdx.x; 
+  while (shmem_loading_idx < blockDim.x * dim) {
+    if (starting_row * dim + shmem_loading_idx < n_rows * dim)
+      dataset_shared[shmem_loading_idx] = dataset[starting_row * dim + shmem_loading_idx];
+    shmem_loading_idx += blockDim.x; 
+  }
+
+  __syncthreads(); 
+
+  if (curr_row < n_rows) {
+    DataT min_distance = std::numeric_limits<DataT>::max(); 
+    IdxT location = 0; 
+    #pragma unroll 16
+    for (int curr_n = 0; curr_n < n_clusters; curr_n++) {
+      DataT curr_distance = dataset_norm[curr_row] + centers_norm[curr_n]; 
+      #pragma unroll 2
+      for (int curr_k = 0; curr_k < dim; curr_k++) {
+        curr_distance -= 2 * dataset_shared[threadIdx.x * dim + curr_k] * centers_shared[curr_n * dim + curr_k]; 
+        /**
+        * Self-neighboring points should have (aNorm == bNorm) == accVal and the dot product (accVal)
+        * can sometimes have round-off errors, which will cause (aNorm == bNorm) ~ accVal instead.
+        */
+        curr_distance = curr_distance * !((curr_distance * curr_distance < raft::distance::detail::ops::get_clamp_precision<DataT>()) * (dataset_norm[curr_row] == centers_norm[curr_n]));
+        if (sqrt) {
```

Comment:
Change it to `constexpr` to make it more clear for both the reader and the compiler
```suggestion
        if constexpr (Sqrt) {
```